libraries/botframework-expressions/src/formatConverter.ts

```diff
@@ -0,0 +1,497 @@
+enum State {
+    None,
+    LowerD1,
+    LowerD2,
+    LowerD3,
+    LowerD4,
+    LowerF1,
+    LowerF2,
+    LowerF3,
+    LowerF4,
+    LowerF5,
+    LowerF6,
+    LowerF7,
+    CapitalF1,
+    CapitalF2,
+    CapitalF3,
+    CapitalF4,
+    CapitalF5,
+    CapitalF6,
+    CapitalF7,
+    LowerG,
+    LowerH1,
+    LowerH2,
+    CapitalH1,
+    CapitalH2,
+    CapitalK,
+    LowerM1,
+    LowerM2,
+    CapitalM1,
+    CapitalM2,
+    CapitalM3,
+    CapitalM4,
+    LowerS1,
+    LowerS2,
+    LowerT1,
+    LowerT2,
+    LowerY1,
+    LowerY2,
+    LowerY3,
+    LowerY4,
+    LowerY5,
+    LowerZ1,
+    LowerZ2,
+    LowerZ3,
+    InSingleQuoteLiteral,
+    InDoubleQuoteLiteral,
+    EscapeSequence,
+}
+
+export function convertCSharpDateTimeToMomentJS(fmtString: string): string {
+    let fmtResult = '';
+    let fmtState: any = State.None;
+    const fmtTolerant = true;
```

Comment:
`fmtTolerant`  is always true, and can not be modified. So, should we remove the logic that when `fmtTolerant` is false?